### PR TITLE
Update docker file to ubuntu22.04 to fix the issue of test_ovs_dpdk

### PR DESCRIPTION
**Pull request by peng6662001**
https://github.com/cloud-hypervisor/cloud-hypervisor/discussions/5402
The test_ovs_dpdk stuck on ubuntu20.04 in docker,but it works fine on ubuntu22.04 in docker.
This patch passed all the integration test case on Ampere Altra when CONFIG_ARM64_4K_PAGES=y for kernel config.

### CODE REVIEW

File: scripts/run_integration_tests_aarch64.sh [hunk 1]
```diff
@@ -11,15 +11,15 @@ build_spdk_nvme() {
     SPDK_DIR="$WORKLOADS_DIR/spdk"
     SPDK_REPO="https://github.com/spdk/spdk.git"
     SPDK_DEPLOY_DIR="/usr/local/bin/spdk-nvme"
-    checkout_repo "$SPDK_DIR" "$SPDK_REPO" master "6301f8915de32baed10dba1eebed556a6749211a"
+    checkout_repo "$SPDK_DIR" "$SPDK_REPO" master "ceefb46358f845d6c4e9f9e18d02ec20a7098ff7"
```
Comment (michael2012z) [inline scripts/run_integration_tests_aarch64.sh:14]: The problem you are resolving is about DPDK, but why do you modify `SPDK`?

File: scripts/run_integration_tests_aarch64.sh [hunk 2]
```diff
@@ -30,13 +30,14 @@ build_spdk_nvme() {
...
     cp -r "$WORKLOADS_DIR/spdk/scripts/rpc" $SPDK_DEPLOY_DIR/rpc
+    cp -r "$WORKLOADS_DIR/spdk/python" $SPDK_DEPLOY_DIR/../
 }
 
 build_virtiofsd() {
     VIRTIOFSD_DIR="$WORKLOADS_DIR/virtiofsd_build"
     VIRTIOFSD_REPO="https://gitlab.com/virtio-fs/virtiofsd.git"
 
-    checkout_repo "$VIRTIOFSD_DIR" "$VIRTIOFSD_REPO" v1.1.0 "220405d7a2606c92636d31992b5cb3036a41047b"
+    checkout_repo "$VIRTIOFSD_DIR" "$VIRTIOFSD_REPO" v1.6.1 "7886ab7c819fd8de370d5a3bbfaae44de56923ed"
```
Comment (michael2012z) [inline scripts/run_integration_tests_aarch64.sh:40]: Is this related?

File: scripts/sha1sums-aarch64 [hunk 3]
```diff
@@ -1,6 +1,6 @@
 6fee67adbfed8db7a225be23ee9d90b5bd7f19e6  bionic-server-cloudimg-arm64.img
 786fe1c33588334e92b35c65e414da068df180bc  bionic-server-cloudimg-arm64.raw
-6e66f9f4b01adc72c884c1c1111e60afadc9c871  bionic-server-cloudimg-arm64.qcow2
+4318bdd54228299259640ba255423c5c4fb90425  bionic-server-cloudimg-arm64.qcow2
```
Comment (michael2012z) [inline scripts/sha1sums-aarch64:3]: I think the SHA shouldn't be changed.

**Review by michael2012z — COMMENTED** [review]
Sorry, I don't understand how the changes in the MR resolve a problem of OVS DPDK.

My suggestion is: It's better to investigate the root cause of the problem of DPDK in your environment, or at least identify some software components that's related to the problem.

I am not sure there is a problem in the 20.04 container. After all, the test works well on the CI server (same type as yours) and on the servers of the other developers.

File: scripts/sha1sums-aarch64 (hunk 3 above)
Comment (peng6662001) [reply scripts/sha1sums-aarch64:3]: In docker of ubuntu20.04,qemu-img --version 
```
qemu-img version 4.2.1 (Debian 1:4.2-3ubuntu6.24)
Copyright (c) 2003-2019 Fabrice Bellard and the QEMU Project developers
```

In docker of ubuntu22.04,qemu-img --version
```
qemu-img version 6.2.0 (Debian 1:6.2+dfsg-2ubuntu6.8)
Copyright (c) 2003-2021 Fabrice Bellard and the QEMU Project developers
```
qemu-img convert -p -f raw -O qcow2 bionic-server-cloudimg-arm64.raw /root/workloads/bionic-server-cloudimg-arm64.qcow2
```
    (100.00/100%) 
```
sha1sum bionic-server-cloudimg-arm64.qcow2 
```
4318bdd54228299259640ba255423c5c4fb90425  bionic-server-cloudimg-arm64.qcow2
```

File: scripts/run_integration_tests_aarch64.sh (hunk 1 above)
Comment (peng6662001) [reply scripts/run_integration_tests_aarch64.sh:14]: There are some error when build on old version of spdk,I don't know how to fix it.
```
/root/workloads/spdk/libvfio-user/samples/client.c: In function 'fake_guest':
/root/workloads/spdk/libvfio-user/samples/client.c:831:9: error: 'MD5_Init' is deprecated: Since OpenSSL 3.0 [-Werror=deprecated-declarations]
  831 |         MD5_Init(&md5_ctx);
      |         ^~~~~~~~
```

File: scripts/run_integration_tests_aarch64.sh (hunk 2 above)
Comment (peng6662001) [reply scripts/run_integration_tests_aarch64.sh:40]: @michael2012z 
If I use the old version of virtiofsd
```
Error running ch-remote command: "target/aarch64-unknown-linux-gnu/release/ch-remote" "--api-socket" "/tmp/ch40QpPE/cloud-hypervisor.sock" "resume"
stderr: Error running command: Server responded with an error: InternalServerError: ApiError(VmResume(Resume(Resume(Invalid transition: InvalidStateTransition(Running, Running)))))

failures:
    common_parallel::test_virtio_fs
    common_parallel::test_virtio_fs_hotplug
    common_parallel::test_virtio_fs_multi_segment                                                        
    common_parallel::test_virtio_fs_multi_segment_hotplug
```

**Review by likebreath — CHANGES_REQUESTED** [review]
I have the same sentiment with @michael2012z that this PR brings too many changes for different reasons. Particularly, moving the base docker image from focal (20.04) to jammy (22.04) itself a major work and should be done in a dedicate PR. 

I would suggest you focus on moving the container base image to jammy, if you believe it aligns with the problem you are solving. Please refer this PR #5072 for potential issues you may encounter. Of course, we are here to support you. It would be a good contribution to the community. Thank you.
